crates/circuit/src/operations.rs

```diff
@@ -305,13 +303,13 @@ static STANDARD_GATE_NAME: [&str; STANDARD_GATE_SIZE] = [
     "cu",           // 39
     "cu1",          // 40
     "cu3",          // 41
-    "c3x",          // 42
+    "c3x",          // 42 ("mcx")
```

Comment:
I feel like this mismatch will cause a problem eventually. Like it's not an issue right now because we never use these gates out of the box anywhere. But we use these names as the canonical identifier in a lot of places (which is why the overloaded `mcx` name is a real problem). For example, when we port the qasm exporters to rust they'll just call `op.name()` internally and then this will return `c3x` instead of `mcx` which will cause issues.

Reply:
I agree. But it seems like the solution would be not using the overloaded `mcx` name in Python, wouldn't it?

Reply:
It would, and ideally that's how I would expect us to fix this. But when we discussed potentially doing that as part of #12628 I thought the concern was around backwards compatibility and the consensus was we shouldn't change the name until 2.0?

Reply:
That's what I vaguely remember. If we want to rename the gates for 2.0, should we then introduce deprecation warnings in 1.2? I am not sure what the protocol is with name changes, we can't add the new path in advance, right?

Reply:
hah, this just became a problem after the recent commits: https://github.com/Qiskit/qiskit/actions/runs/9764823435/job/26954059268?pr=12659#step:8:16894

Reply:
The deprecation path for something like this isn't clear as it's a straight change in behavior. There isn't a good way to signal this to users unless we emit a `FutureWarning` every time a `C3XGate` is created. This might be the case we just document this in the 2.0 release notes and say it wasn't easy to raise a deprecation warning for.

Reply:
Ok, I didn't see a way around the recent failures other than renaming "c3x" to "mcx" in rust space ([f05c9a1](https://github.com/Qiskit/qiskit/pull/12659/commits/f05c9a1b320ad61e79ecc711e66208de083bc11a)). I don't love it, but I guess we can then change both names at the same time?